src/linear_algebra/matrix/schur_complement.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2022 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alexander Bentkamp, Jeremy Avigad, Johan Commelin
+-/
+import linear_algebra.matrix.symmetric
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.matrix.pos_def
+
+/-! # Schur complement
+
+This file proves properties of the Schur complement `D - C A⁻¹ B` of a block matrix `[A B; C D]`.
```

Comment:
I think it would be good to add a reference to [`matrix.det_from_blocks₁₁`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/nonsingular_inverse.html#matrix.det_from_blocks%E2%82%81%E2%82%81) here since that lives in another file.